services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,753 @@
+            return resourceUrl
+                   + (resourceUrl.indexOf('?') >= 0 ? "&" : "?")
+                   + "Expires=" + expirationDate.getEpochSecond()
+                   + "&Signature=" + urlSafeSignature
+                   + "&Key-Pair-Id=" + keyPairId;
+        } catch (InvalidKeyException e) {
+            throw SdkClientException.create("Could not sign url", e);
+        }
+    }
+
+    /**
+     * Returns a signed URL that provides tailored access to private content
+     * based on an access time window and an ip range.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-creating-signed-url-custom-policy.html"
+     * >Creating a signed URL using a custom policy</a>.
+     *
+     * @param protocol
+     *            The protocol of the URL - HTTP or HTTPS
+     * @param distributionDomain
+     *            The domain name of the distribution
+     * @param s3ObjectKey
+     *            The s3 key of the object
+     * @param privateKeyFile
+     *            Your private key file. RSA private key (.pem) and pkcs8 (.der)
+     *            files are supported.
+     * @param keyPairId
+     *            Identifier of a public/private certificate keypair already
+     *            configured in your Amazon Web Services account.
+     * @param activeDate
+     *            The beginning valid date of the signed URL in UTC
+     * @param expirationDate
+     *            The expiration date of the signed URL in UTC
+     * @param ipRange
+     *            The allowed IP address range of the client making the GET
+     *            request, in CIDR form (e.g. 192.168.0.1/24).
+     * @return The signed URL.
+     */
+    public static String getSignedUrlWithCustomPolicy(Protocol protocol,
+                                                      String distributionDomain,
+                                                      String s3ObjectKey,
+                                                      File privateKeyFile,
+                                                      String keyPairId,
+                                                      Instant activeDate,
+                                                      Instant expirationDate,
+                                                      String ipRange) throws InvalidKeySpecException, IOException {
+        PrivateKey privateKey = loadPrivateKey(privateKeyFile);
+        String resourceUrl = generateResourceUrl(protocol, distributionDomain, s3ObjectKey);
+        String policy = buildCustomPolicyForSignedUrl(resourceUrl, activeDate, expirationDate, ipRange);
+        return getSignedUrlWithCustomPolicy(resourceUrl, privateKey, keyPairId, policy);
+    }
+
+    /**
+     * Generate a signed URL that allows access to distribution and S3 objects
+     * by applying access restrictions specified in a custom policy document.
```

Comment:
It can be helpful to include that the custom policy itself is included as a part of the presigned url (and for the other method, that it's not).

Reply:
Added description to mention this